units/patch_intent.go

```diff
@@ -1258,16 +1258,33 @@ func (j *patchIntentProcessor) sendGitHubErrorStatus(ctx context.Context, patchD
 	j.AddError(update.Error())
 }
 
-// sendGitHubSuccessMessage sends a successful status to Github with the given message.
-func (j *patchIntentProcessor) sendGitHubSuccessMessage(ctx context.Context, patchDoc *patch.Patch, msg string) {
-	update := NewGithubStatusUpdateJobWithSuccessMessage(
-		evergreenContext,
-		patchDoc.GithubPatchData.BaseOwner,
-		patchDoc.GithubPatchData.BaseRepo,
-		patchDoc.GithubPatchData.HeadHash,
-		msg,
-	)
-	update.Run(ctx)
-
-	j.AddError(update.Error())
+// sendGitHubSuccessMessages sends a successful status to Github with the given message for all
+// branch protection rules configured for the given project.
+func (j *patchIntentProcessor) sendGitHubSuccessMessages(ctx context.Context, patchDoc *patch.Patch, projectRef *model.ProjectRef, msg string) {
+	rules, err := thirdparty.GetEvergreenBranchProtectionRules(ctx, "", projectRef.Owner, projectRef.Repo, projectRef.Branch)
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "getting branch protection rules for %s/%s/%s", projectRef.Owner, projectRef.Repo, projectRef.Branch))
+		return
+	}
+	// If the base evergreen context is not one of the rules, add it.
+	hasEvergreenContext := false
+	for _, rule := range rules {
+		if rule == evergreenContext {
+			hasEvergreenContext = true
+		}
+	}
+	if !hasEvergreenContext {
+		rules = append(rules, evergreenContext)
```

Comment:
another option is `utility.UniqueStrings(append(rules, evergreenContext))` to drop this to one line; we do this pretty frequently when we don't expect the list to be long.